### PR TITLE
Defines more area and volume calculations in `Operations`

### PR DESCRIPTION
**Pull request by tomchor**
Defines and exports areas `Axᶜᶠᶠ`, `Axᶠᶠᶠ`, `Ayᶠᶜᶠ` and `Ayᶠᶠᶠ` and volumes `Vᶠᶠᶜ`, `Vᶠᶜᶠ`, `Vᶜᶠᶠ`, `Vᶠᶠᶠ` in `Operations`.

Closes https://github.com/CliMA/Oceananigans.jl/issues/2189

### CODE REVIEW

**Comment by tomchor** [comment]
I'm also bumping the version since this is a bug fix.

File: src/Operators/spacings_and_areas_and_volumes.jl [hunk 1]
```diff
@@ -92,17 +92,22 @@ using Oceananigans.Grids: Flat
...
+@inline Axᶠᶜᶠ(i, j, k, grid::Union{ARG, AHCG}) = Δyᶠᶜᵃ(i, j, k, grid) * Δzᵃᵃᶠ(i, j, k, grid) # η
+@inline Axᶜᶠᶠ(i, j, k, grid::Union{ARG, AHCG}) = Δyᶜᶠᵃ(i, j, k, grid) * Δzᵃᵃᶠ(i, j, k, grid)
+@inline Axᶠᶠᶠ(i, j, k, grid::Union{ARG, AHCG}) = Δyᶠᶠᵃ(i, j, k, grid) * Δzᵃᵃᶠ(i, j, k, grid)
+
 
 @inline Ayᶜᶜᶜ(i, j, k, grid::Union{ARG, AHCG}) = Δxᶜᶜᵃ(i, j, k, grid) * Δzᵃᵃᶜ(i, j, k, grid) # c
-@inline Ayᶜᶠᶜ(i, j, k, grid::Union{ARG, AHCG}) = Δxᶜᶠᵃ(i, j, k, grid) * Δzᵃᵃᶜ(i, j, k, grid) # v
 @inline Ayᶠᶜᶜ(i, j, k, grid::Union{ARG, AHCG}) = Δxᶠᶜᵃ(i, j, k, grid) * Δzᵃᵃᶜ(i, j, k, grid) # v
+@inline Ayᶜᶠᶜ(i, j, k, grid::Union{ARG, AHCG}) = Δxᶜᶠᵃ(i, j, k, grid) * Δzᵃᵃᶜ(i, j, k, grid) # v
+@inline Ayᶜᶜᶠ(i, j, k, grid::Union{ARG, AHCG}) = Δxᶜᶜᵃ(i, j, k, grid) * Δzᵃᵃᶠ(i, j, k, grid) # ξ
```
Comment (navidcy) [inline src/Operators/spacings_and_areas_and_volumes.jl:106]: ζ is vertical vorticity, right?
what does ξ stands for?

**Comment by navidcy** [comment]
How about few tests? E.g., some similar to what #2189 is discussion. Would that be useful you reckon?

File: src/Operators/spacings_and_areas_and_volumes.jl (hunk 1 above)
Comment (tomchor) [reply src/Operators/spacings_and_areas_and_volumes.jl:106]: I have no idea. These comments were there before. I didn't comment on any of the lines that I actually added.

Comment (navidcy) [reply src/Operators/spacings_and_areas_and_volumes.jl:106]: no worries then! :)

**Comment by navidcy** [comment]
Nice! I'll have a look in a bit!

**Comment by navidcy** [comment]
Or perhaps when tests pass? Let me know if you need help with that.

**Comment by tomchor** [comment]
> Or perhaps when tests pass? Let me know if you need help with that.

Thanks. I think that test failed due to some server issue. I restarted it and it seems to be going fine.

**Comment by glwagner** [comment]
These have to be defined for all grids...

**Comment by glwagner** [comment]
@tomchor can you list the operators you added in the description of this PR?

**Comment by tomchor** [comment]
> @tomchor can you list the operators you added in the description of this PR?

Done!

> These have to be defined for all grids...

These are defined for `Union{ARG, AHCG}`, which I believe are abstract rectilinear and horizontally curvilinear grids, no? I'd prefer not to define these for other grids in this PR since I have no familiarity with them if that's okay.

**Comment by glwagner** [comment]
> > @tomchor can you list the operators you added in the description of this PR?
> 
> Done!
> 
> > These have to be defined for all grids...
> 
> These are defined for `Union{ARG, AHCG}`, which I believe are abstract rectilinear and horizontally curvilinear grids, no? I'd prefer not to define these for other grids in this PR since I have no familiarity with them if that's okay.

I believe that covers all grids we have, so we are good. The tests should loop over all grids too!

**Comment by glwagner** [comment]
@tomchor can I commit to this PR and add more tests?

**Comment by tomchor** [comment]
> @tomchor can I commit to this PR and add more tests?

Sure!

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
@tomchor, version got bumped some other bug fix PR so I bumped it more!

nice!

File: test/test_abstract_operations.jl [hunk 2]
```diff
@@ -271,21 +271,39 @@ for arch in archs
...
 
         @testset "BinaryOperations with GridMetricOperation [$(typeof(arch))]" begin
-            grid = RectilinearGrid(arch, size=(1, 1, 1), extent=(2, 3, 4))
+            lat_lon_grid = LatitudeLongitudeGrid(arch, size=(1, 1, 1), longitude=(0, 1), latitude=(0, 1), z=(0, 1))
+            rectilinear_grid = RectilinearGrid(arch, size=(1, 1, 1), extent=(2, 3, 4))
+
+            for LX in (Center, Face)
+                for LY in (Center, Face)
+                    for LZ in (Center, Face)
+                        loc = (LX, LY, LZ)
```
Comment (tomchor) [inline test/test_abstract_operations.jl:280]: Kudos for this very thorough testing routine :+1:

**Comment by navidcy** [comment]
@tomchor shall I merge?

**Pull request merged**